src/components/diagrams/diagram-pane.js

```diff
@@ -286,95 +362,314 @@ export function DiagramPane({
+                        ...newDiagram,
+                    };
+                }
+                // otherwise we just add a new one
+                else {
+                    updatedViews.push(newDiagram);
+                }
+                return updatedViews;
+            });
+
+            // Then we add the data when the fetch is finished
+            createView({
                 ids: networkAreaIds,
                 state: networkAreaViewState,
                 svgType: DiagramType.NETWORK_AREA_DIAGRAM,
                 depth: networkAreaDiagramDepth,
+            }).then((networkAreaDiagramView) => {
+                setViews((views) => {
+                    const updatedViews = views.slice();
+                    // the NAD is always in last position
+                    updatedViews[updatedViews.length - 1] = {
+                        ...updatedViews[updatedViews.length - 1],
+                        ...networkAreaDiagramView,
+                        loadingState: false,
+                    };
+                    return updatedViews;
+                });
             });
+        },
+        [createView, intl]
+    );
 
-            // if current view cannot be found, it returns undefined
-            // in this case, we remove all the NAD from diagram states
-            if (networkAreaDiagramView) {
-                diagramViews.push({
-                    ...networkAreaDiagramView,
-                    align: 'right',
-                });
-            } else {
-                closeDiagramView(null, DiagramType.NETWORK_AREA_DIAGRAM); // In this case, the ID is irrelevant
+    const removeNAD = useCallback(() => {
+        setViews((views) => {
+            const updatedViews = views.filter(
+                (view) => view.svgType !== DiagramType.NETWORK_AREA_DIAGRAM
+            );
+            return updatedViews;
+        });
+    }, []);
+
+    const updateNAD = useCallback(
+        (diagramStates) => {
+            const networkAreaIds = [];
+            let networkAreaViewState = ViewState.OPENED;
+            diagramStates.forEach((diagramState) => {
+                if (diagramState.svgType === DiagramType.NETWORK_AREA_DIAGRAM) {
+                    networkAreaIds.push(diagramState.id);
+                    networkAreaViewState = diagramState.state; // They should all be the same value
+                }
+            });
+            if (networkAreaIds.length > 0) {
+                const sameNadAlreadyPresentInViews = viewsRef.current.find(
+                    (diagramView) =>
+                        diagramView.svgType ===
+                            DiagramType.NETWORK_AREA_DIAGRAM &&
+                        diagramView.ids?.toString() ===
+                            networkAreaIds.toString() &&
+                        diagramView.depth === networkAreaDiagramDepth
+                );
+                if (!sameNadAlreadyPresentInViews) {
+                    addOrReplaceNAD(
```

Comment:
The name of variable or functions do not make sense : if the NAD is not already in the views, why should we use a "xxxReplace" function? If the NAD **is** already present in the views, we **should** use a xxxReplace function, therefor why do we test `!sameNadAlreadyPresentInViews` ? Shouldn't we call the function below whether it's true or false ?

Reply:
Because a NAD, but different one (less or more VL inside, or different depth) can be present. In this case we replace it.
The boolean is here to check if we need to update (by adding or replacing) the NAD view.